batchglm/train/numpy/base_glm/estimator.py

```diff
@@ -31,29 +32,63 @@ class EstimatorGlm(_EstimatorGLM, metaclass=abc.ABCMeta):
         A list of all log likelihood updates
     """
 
-    def __init__(
-        self,
-        model: ModelIwls,
-        input_data: InputDataGLM,
-        dtype: str,
-    ):
+    _train_loc: bool = False
+    _train_scale: bool = False
+    _model_container: BaseModelContainer
+    lls: List[float] = []
+    dtype: str = ""
+
+    def __init__(self, dtype: str, provide_batched: bool = False):
         """
         Performs initialisation and creates a new estimator.
         :param model:
             The IWLS model to be fit
-        :param input_data:
-            The input data for the model
         :param dtype:
             i.e float64
         """
-        if input_data.design_scale.shape[1] != 1:
+        if self.model_container.design_scale.shape[1] != 1:
```

Comment:
Not sure how I feel about the parent having to set this in its initializer and then the child (i.e this class) just expecting it to be there.  Maybe we should at least check if this is `None` first?

Reply:
Why would it be None?
It is sth. that needs to be checked for all distribution models which is why I placed it in the base_estimator. The model itself should be fine with having more than an intercept for the scale model. It just cannot be fit with numpy right now.

Reply:
Ok I get that this is a bit weird. I changed how this works. The model_container is now provided to the base class estimator which guarantees it exists.